backend/src/v2/compiler/argocompiler/container.go

```diff
@@ -527,38 +536,61 @@ func (c *workflowCompiler) getTaskRetryPolicySpec(name string) *pipelinespec.Pip
 	return taskSpec.RetryPolicy
 }
 
-func (c *workflowCompiler) getTaskRetryStrategy(name string) *wfapi.RetryStrategy {
+func (c *workflowCompiler) getTaskRetryParameters(name string) *[]wfapi.Parameter {
 	retryPolicy := c.getTaskRetryPolicySpec(name)
 	if retryPolicy == nil {
-		return nil
-	}
+		return &[]wfapi.Parameter{}
+	}
+	// Create slice of parameters with "Name" field set.
+	parameters := []wfapi.Parameter{
+		{Name: paramRetryMaxCount},
+		{Name: paramRetryBackOffDuration},
+		{Name: paramRetryBackOffFactor},
+		{Name: paramRetryBackOffMaxDuration}}
+	return &parameters
+}
 
-	argoBackOffDuration := "0"
-	backoffDuration := retryPolicy.GetBackoffDuration()
-	if backoffDuration != nil {
-		argoBackOffDuration = strconv.FormatInt(backoffDuration.Seconds, 10)
-	}
+func (c *workflowCompiler) getTaskRetryParametersWithValues(name string) *[]wfapi.Parameter {
```

Comment:
We don't need to use a pointer here on the returned slice.
```suggestion
func (c *workflowCompiler) getTaskRetryParametersWithValues(name string) []wfapi.Parameter {
```

Reply:
Updated